lib/engine/game/base.rb

```diff
@@ -1065,6 +1065,12 @@ def cache_objects
       def bankruptcy_limit_reached?
         @players.any?(&:bankrupt)
       end
+
+      def par_prices_for_entity(entity: current_entity)
+        @stock_market.par_prices.map do |p|
```

Comment:
you put this method in the private section it should be above that

also this method would be cleaner as

par_prices.select { |p| p.price * 2 <= entity.cash }